docs/content/ingestion/stream-pull.md

```diff
@@ -335,3 +335,18 @@ This can be difficult to manage at scale, especially with multiple partitions.
 
 Each standalone realtime node has its own set of logs. Diagnosing errors across many partitions across many servers may be 
 difficult to manage and track at scale.
+
+## Deployment Notes
+
+Stream ingestion may generate a large number of small segments because it's difficult to optimize the segment size at
+ingestion time. The number of segments will increase over time, and this might cuase the query performance issue. 
+
+To mitigate this problem, it is recommended that scheduled re-indexing
```

Comment:
this blurb appears to be repeated in multiple places.
how about creating a separate page that talks about segment compaction options available in Druid and linking to that page from all different pages?

Reply:
I guess you meant the paragraph about the needs for reindexing for steaming ingestion is repeated in several places because new sections for describing segment compaction are already added (https://github.com/druid-io/druid/pull/5102/files#diff-1401189ed202eac3beb0b4e0b4d535cbR41, https://github.com/druid-io/druid/pull/5102/files#diff-ad94e83d5b5a9cf743a09e7ccb0d9a95R127).

But, this looks a quite short paragraph (like only 6 lines) to create a new separate page. Do you think it's worthwhile?

Reply:
I was writing same comments in multiple places and thought maybe a separate page just talking about how to maintain segments of optimal size and what causes small segments etc etc could be created. i don't have any super strong feelings about it.

Reply:
Sounds good. I added a new document.